src/main/java/com/mparticle/kits/RokoMobiKit.java

```diff
@@ -0,0 +1,260 @@
+                            .setLink(intent.getDataString())
+                            .setServiceProviderId(getConfiguration().getKitId())
+                            .setParameters(new JSONObject(Json.serialize(res.data)));
+                    getKitManager().onResult(deepLinkResult);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void failure(final String error) {
+                getKitManager().onError(new DeepLinkError()
+                        .setMessage(error)
+                        .setServiceProviderId(getConfiguration().getKitId()));
+
+            }
+        });
+    }
+
+    @Override
+    public void setInstallReferrer(Intent intent) {
+        RokoLinks.handleDeepLink(intent);
+    }
+
+    @Override
+    public boolean willHandlePushMessage(Intent intent) {
+        return intent != null && intent.getStringExtra(RK_MSG_ID) != null;
+    }
+
+    @Override
+    public void onPushMessageReceived(final Context context, Intent intent) {
+        final Bundle bundle = intent.getExtras();
+        ThreadUtils.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if (bundle.containsKey(RokoPushConstants.EXTRA_OVERLAY_ID)) {
+                    PushNotificationHelper.pulNotificationSent(bundle.getString(RokoPushConstants.EXTRA_OVERLAY_ID));
+                    PushNotificationHelper.showDefaultNotification(context, bundle);
+                } else if (bundle.containsKey(RokoPushConstants.EXTRA_CONVERSATION_ID)) {
+                    PushNotificationHelper.showInstabot(bundle.getString(RokoPushConstants.EXTRA_CONVERSATION_ID));
+                } else {
+                    PushNotificationHelper.showDefaultNotification(context, bundle);
+                }
+            }
+        });
+    }
+
+    @Override
+    public boolean onPushRegistration(String instanceId, String senderId) {
+        Settings preferences = RokoMobi.getSettings();
+        preferences.edit().putString(DEVICE_TOKEN, instanceId).apply();
+        RokoPush.register(getContext(), instanceId);
+        RokoPush.start(senderId);
+        return false;
+    }
+
+
+    @Override
+    public void setUserAttribute(String name, String value) {
+        RokoAccount.setUserCustomProperty(name, value);
+    }
+
+    @Override
+    public void setUserAttributeList(String s, List<String> list) {
+    }
+
+    @Override
+    public boolean supportsAttributeLists() {
+        return false;
+    }
+
+    @Override
+    public void setAllUserAttributes(Map<String, String> map, Map<String, List<String>> map1) {
+    }
+
+    @Override
+    public void removeUserAttribute(String name) {
+        RokoAccount.setUserCustomProperty(name, null);
+    }
+
+    @Override
+    public void setUserIdentity(MParticle.IdentityType identityType, String identityToken) {
+        RokoAccount.setUser(identityToken);
```

Comment:
You likely only want to do this for customer ID or email identitytypes. mParticle customers often set both, in addition to other IDs.

Reply:
agreed, we'll update PR to add logic for setting only  customer ID or email identitytypes